animation_nodes/math/conversion.pyx

```diff
@@ -120,7 +122,20 @@ cdef toPyEuler3(Euler3* e):
     if e.order == 3: return Euler((e.x, e.y, e.z), "YZX")
     if e.order == 4: return Euler((e.x, e.y, e.z), "ZXY")
     if e.order == 5: return Euler((e.x, e.y, e.z), "ZYX")
-
+    
+#https://en.wikipedia.org/wiki/Conversion_between_quaternions_and_Euler_angles
+cdef euler3ToQuaternion(Quaternion* q, Euler3 *e):
+    cdef float cx = cos(e.x * 0.5)
+    cdef float sx = sin(e.x * 0.5)
+    cdef float cy = cos(e.y * 0.5)
+    cdef float sy = sin(e.y * 0.5)
+    cdef float cz = cos(e.z * 0.5)
+    cdef float sz = sin(e.z * 0.5)
+    
+    q.w = cx * cy * cz + sx * sy * sz
+    q.x = sx * cy * cz - cx * sy * sz
+    q.y = cx * sy * cz + sx * cy * sz
+    q.z = cx * cy * sz - sx * sy * cz
```

Comment:
Looks like you have some common expressions here, try to group them. This applies to the rest of the functions as well.

Reply:
sorry, i don't fully understand how to group them. i guess like this:
`cdef float cxcy = cx * cy`
 `cdef float sxcy = sx * cy`
 `cdef float cxsy = cx * sy`
` cdef float sxsy = sx * sy`

`q.w = cxcy * cz + sxsy *sz`
`q.x =  sxcy * cz - cxsy * sz`
`q.y = cxsy * cz + sxsy * sz`
`q.z = cxcy * sz -  sxsy *cz`

Reply:
@kdsin23 Yes, exactly.

Reply:
Thank you!

Reply:
BTW, is there a handy way to move all of those functions to `animation_nodes/math/rotation_conversion`?

Reply:
Handy as in automatic? Not that I know of.